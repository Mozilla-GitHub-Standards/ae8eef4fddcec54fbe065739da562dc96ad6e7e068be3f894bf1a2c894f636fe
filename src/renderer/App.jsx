/** @format */

import React, { Component } from 'react'

import MainLayout from './layouts/MainLayout'

export default class App extends Component {
  render() {
    return <MainLayout />
  }
}
