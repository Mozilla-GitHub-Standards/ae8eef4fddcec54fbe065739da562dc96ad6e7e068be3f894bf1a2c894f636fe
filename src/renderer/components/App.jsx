/**
 * @format
 * @flow
 */

import React from 'react'
import { Link } from 'react-router-dom'
import Button from '@material-ui/core/Button'
import { remote } from 'electron'
import styled, { keyframes, createGlobalStyle } from 'styled-components'

import logo from '../images/logo.svg'

const GlobalStyle = createGlobalStyle`
  body {
    margin: 0px;
    padding: 0px;
    font-family: sans-serif;
    background-color: #303341;
  }
`

const AppContainer = styled.div`
  text-align: center;
`

const AppIntro = styled.p`
  font-size: large;
`

const AppHeader = styled.header`
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
`

const Title = styled.h1`
  font-size: 1.5em;
`

const LogoSpinner = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`

const Logo = styled.img`
  animation: ${LogoSpinner} infinite 20s linear;
  height: 80px;
`

const GradientButton = {
  background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
  borderradius: 3,
  border: 0,
  color: 'white',
  height: 48,
  padding: '0 30px',
  boxshadow: '0 3px 5px 2px rgba(255, 105, 135, .3)'
}

export default class App extends React.Component<{}, {}> {
  render() {
    return (
      <React.Fragment>
        <GlobalStyle />
        <AppContainer>
          <AppHeader>
            <Logo src={logo} />
            <Title>Welcome to React in Electron packed with Parcel!</Title>
            Electron App Version: {remote.app.getVersion()}
          </AppHeader>
          <AppIntro>
            <Button component={Link} to="/" style={GradientButton}>
              Back
            </Button>
          </AppIntro>
        </AppContainer>
      </React.Fragment>
    )
  }
}
