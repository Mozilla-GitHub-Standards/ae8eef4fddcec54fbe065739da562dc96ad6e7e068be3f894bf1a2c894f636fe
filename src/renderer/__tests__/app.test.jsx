/** @format */

import React from 'react'
import Renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

import App from '../components/App'

test('Component: App', () => {
  const component = Renderer.create(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
