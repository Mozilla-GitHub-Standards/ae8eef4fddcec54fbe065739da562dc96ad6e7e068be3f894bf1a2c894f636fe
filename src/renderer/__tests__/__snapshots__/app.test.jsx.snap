// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App 1`] = `
<div
  className="App__AppContainer-sc-1gqgb50-0 fRSoNO"
>
  <header
    className="App__AppHeader-sc-1gqgb50-2 kiObkE"
  >
    <img
      className="App__Logo-sc-1gqgb50-4 jnWBhu"
      src="logo.svg"
    />
    <h1
      className="App__Title-sc-1gqgb50-3 fzEvPq"
    >
      Welcome to React in Electron packed with Parcel!
    </h1>
    Electron App Version: 
    2.0.9
  </header>
  <p
    className="App__AppIntro-sc-1gqgb50-1 dOIlSn"
  >
    <a
      className="MuiButtonBase-root-027 MuiButton-root-01 MuiButton-text-03 MuiButton-flat-06"
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={
        Object {
          "background": "linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)",
          "border": 0,
          "borderradius": 3,
          "boxshadow": "0 3px 5px 2px rgba(255, 105, 135, .3)",
          "color": "white",
          "height": 48,
          "padding": "0 30px",
        }
      }
      tabIndex="0"
    >
      <span
        className="MuiButton-label-02"
      >
        Back
      </span>
      <span
        className="MuiTouchRipple-root-030"
      />
    </a>
  </p>
</div>
`;
